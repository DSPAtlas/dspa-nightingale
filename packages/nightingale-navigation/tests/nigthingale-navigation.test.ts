import NightingaleNavigation from "../dist/index";

let rendered: NightingaleNavigation;

// eslint-disable-next-line jest/no-disabled-tests
describe.skip("nightingale-navigation tests", () => {
  beforeEach(() => {
    rendered = new NightingaleNavigation();
    rendered.setAttribute("height", "15");
    rendered.setAttribute("length", "10");
    rendered.setAttribute("display-start", "1");
    rendered.setAttribute("display-end", "1");
    document.documentElement.appendChild(rendered);
  });

  afterEach(() => {
    document.documentElement.removeChild(rendered);
  });

  test("it should display the navigation correctly", async () => {
    expect(rendered).toMatchSnapshot();
  });

  test("it should display the navigation correctly after zoom", () => {
    rendered.setAttribute("display-start", "2");
    rendered.setAttribute("display-end", "4");
    return new Promise((done) => {
      window.requestAnimationFrame(() => {
        expect(rendered).toMatchSnapshot();
        done(true);
      });
    });
  });
});
