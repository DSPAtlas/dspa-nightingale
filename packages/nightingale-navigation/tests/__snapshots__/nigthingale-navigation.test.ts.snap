// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nightingale-navigation tests it should display the navigation correctly 1`] = `
<nightingale-navigation
  height="15"
  style="display: inline-block;"
>
  <!---->
  <svg
    class="container"
    height="15"
    id=""
    width="800"
  >
    <text
      class="start-label"
      style="font-family: sans-serif;"
      x="10"
      y="13"
    >
      1
    </text>
    <text
      class="end-label"
      style="font-family: sans-serif;"
      text-anchor="end"
      x="790"
      y="13"
    >
      1
    </text>
    <g
      class="x axis"
      fill="none"
      font-family="sans-serif"
      font-size="10"
      text-anchor="middle"
    >
      <path
        class="domain"
        d="M10.5,6V0.5H790.5V6"
        stroke="currentColor"
      />
      <g
        class="tick"
        opacity="1"
        transform="translate(400.5,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          1
        </text>
      </g>
    </g>
    <g
      class="brush"
      fill="none"
      pointer-events="all"
    >
      <rect
        class="overlay"
        cursor="crosshair"
        height="10.5"
        pointer-events="all"
        width="780"
        x="10"
        y="0"
      />
      <rect
        class="selection"
        cursor="move"
        fill="#777"
        fill-opacity="0.3"
        height="10.5"
        shape-rendering="crispEdges"
        stroke="#fff"
        style=""
        width="0"
        x="400"
        y="0"
      />
      <rect
        class="handle handle--w"
        cursor="ew-resize"
        height="16.5"
        style=""
        width="6"
        x="397"
        y="-3"
      />
      <rect
        class="handle handle--e"
        cursor="ew-resize"
        height="16.5"
        style=""
        width="6"
        x="397"
        y="-3"
      />
    </g>
    <polygon
      class="zoom-polygon"
      fill="#777"
      fill-opacity="0.3"
      points="400,7.5
        400,7.5
        790,15
        10,15"
    />
    <g
      class="margin"
    >
      <rect
        class="margin-left"
        fill="#FFFFFFDD"
        height="15"
        style="pointer-events: none;"
        width="10"
        x="0"
      />
      <rect
        class="margin-right"
        fill="#FFFFFFDD"
        height="15"
        style="pointer-events: none;"
        width="10"
        x="790"
      />
      <rect
        class="margin-top"
        fill="#FFFFFFDD"
        height="0"
        style="pointer-events: none;"
        width="800"
        x="0"
        y="0"
      />
      <rect
        class="margin-bottom"
        fill="#FFFFFFDD"
        height="0"
        style="pointer-events: none;"
        width="800"
        x="0"
        y="15"
      />
    </g>
    <g
      class="highlighted"
    />
  </svg>
</nightingale-navigation>
`;

exports[`nightingale-navigation tests it should display the navigation correctly after zoom 1`] = `
<nightingale-navigation
  display-end="4"
  display-start="2"
  height="15"
  style="display: inline-block;"
>
  <!---->
  <svg
    class="container"
    height="15"
    id=""
    width="800"
  >
    <text
      class="start-label"
      style="font-family: sans-serif;"
      x="10"
      y="13"
    >
      1
    </text>
    <text
      class="end-label"
      style="font-family: sans-serif;"
      text-anchor="end"
      x="790"
      y="13"
    >
      1
    </text>
    <g
      class="x axis"
      fill="none"
      font-family="sans-serif"
      font-size="10"
      text-anchor="middle"
    >
      <path
        class="domain"
        d="M10.5,6V0.5H790.5V6"
        stroke="currentColor"
      />
      <g
        class="tick"
        opacity="1"
        transform="translate(400.5,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          1
        </text>
      </g>
    </g>
    <g
      class="brush"
      fill="none"
      pointer-events="all"
    >
      <rect
        class="overlay"
        cursor="crosshair"
        height="10.5"
        pointer-events="all"
        width="780"
        x="10"
        y="0"
      />
      <rect
        class="selection"
        cursor="move"
        fill="#777"
        fill-opacity="0.3"
        height="10.5"
        shape-rendering="crispEdges"
        stroke="#fff"
        style=""
        width="0"
        x="400"
        y="0"
      />
      <rect
        class="handle handle--w"
        cursor="ew-resize"
        height="16.5"
        style=""
        width="6"
        x="397"
        y="-3"
      />
      <rect
        class="handle handle--e"
        cursor="ew-resize"
        height="16.5"
        style=""
        width="6"
        x="397"
        y="-3"
      />
    </g>
    <polygon
      class="zoom-polygon"
      fill="#777"
      fill-opacity="0.3"
      points="400,7.5
        400,7.5
        790,15
        10,15"
    />
    <g
      class="margin"
    >
      <rect
        class="margin-left"
        fill="#FFFFFFDD"
        height="15"
        style="pointer-events: none;"
        width="10"
        x="0"
      />
      <rect
        class="margin-right"
        fill="#FFFFFFDD"
        height="15"
        style="pointer-events: none;"
        width="10"
        x="790"
      />
      <rect
        class="margin-top"
        fill="#FFFFFFDD"
        height="0"
        style="pointer-events: none;"
        width="800"
        x="0"
        y="0"
      />
      <rect
        class="margin-bottom"
        fill="#FFFFFFDD"
        height="0"
        style="pointer-events: none;"
        width="800"
        x="0"
        y="15"
      />
    </g>
    <g
      class="highlighted"
    />
  </svg>
</nightingale-navigation>
`;
